src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -483,7 +483,7 @@ protected SyntaxToken EatToken(SyntaxKind kind)
         }
 
         // Consume a token if it is the right kind. Otherwise skip a token and replace it with one of the correct kind.
-        protected SyntaxToken EatTokenAsKind(SyntaxKind expected)
+        protected SyntaxToken EatTokenAsKind(SyntaxKind expected, bool reportError = true)
```

Comment:
i don't see any place (currently) where EatTokenAsKind is called with `reportError: false`.  Is this needed?